src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -250,5 +235,36 @@ public static string GetGraphQLTypeForColumnType(Type type)
                         subStatusCode: DataApiBuilderException.SubStatusCodes.GraphQLMapping)
             };
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="metadataValue"></param>
+        /// <returns></returns>
+        /// <exception cref="DataApiBuilderException"></exception>
+        public static IValueNode CreateValueNodeFromDbObjectMetadata(object metadataValue)
+        {
+            IValueNode arg = metadataValue switch
+            {
+                byte value => new ObjectValueNode(new ObjectFieldNode(BYTE_TYPE, new IntValueNode(value))),
+                short value => new ObjectValueNode(new ObjectFieldNode(SHORT_TYPE, new IntValueNode(value))),
+                int value => new ObjectValueNode(new ObjectFieldNode(INT_TYPE, value)),
+                long value => new ObjectValueNode(new ObjectFieldNode(LONG_TYPE, new IntValueNode(value))),
```

Comment:
Is this potentially a loss of precision from long to IntValueNode?

Reply:
IntValueNode is a hotChocolate type that encapsulates integral numeric types: https://github.com/ChilliCream/graphql-platform/blob/12.18.0/src/HotChocolate/Language/src/Language.SyntaxTree/IntValueNode.cs

So it has constructors and fields which hold those types: e.g. byte, short, int, long, etc. Precision is maintained. It is not immediately obvious looking at this code only. Worth adding a comment to the code on my part.

Reply:
Yeah, a comment would help here.

Reply:
Added comment and also noting that this method wasn't changed for this PR, was only put into a separate helper function.